core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -186,4 +201,43 @@ ManifestFile manifest() {
       return manifest;
     }
   }
+
+  static class ContentFileStructWithMetrics implements StructLike {
+    private final Schema projectedFilesSchema;
```

Comment:
I'd probably just define a variable called `size` instead of referencing the complete schema.

Overall, `StructLike` is serializable so we have to be careful with what we put into fields. Serializing an int will be much more efficient and would work across different serialization mechanisms.